scripts/tests/yaml/tests_logger.py

```diff
@@ -0,0 +1,417 @@
+
+    def __print_results(self, logger):
+        for entry in logger.entries:
+            if entry.is_warning():
+                state = _WARNING
+            elif entry.is_error():
+                state = _FAILURE
+            else:
+                state = ' '  # Do not mark success to not make the output hard to read
+
+            print(self.__strings.result_entry.format(state=state, category=entry.category, message=entry.message))
+
+    def __print_failure(self, expected_response, received_response):
+        expected_response = self.__prepare_data_for_printing(expected_response)
+        expected_value = json.dumps(
+            expected_response, sort_keys=True, indent=2, separators=(',', ': '))
+        expected_value = expected_value.replace('\n', '\n\t\t              ')
+
+        received_response = self.__prepare_data_for_printing(
+            received_response)
+        received_value = json.dumps(
+            received_response, sort_keys=True, indent=2, separators=(',', ': '))
+        received_value = received_value.replace('\n', '\n\t\t               ')
+        print(self.__strings.result_failure.format(key=click.style("Expected", bold=True), value=expected_value))
+        print(self.__strings.result_failure.format(key=click.style("Received", bold=True), value=received_value))
+
+    def __prepare_data_for_printing(self, data):
+        if isinstance(data, bytes):
+            return data.decode('unicode_escape')
+        elif isinstance(data, list):
+            return [self.__prepare_data_for_printing(entry) for entry in data]
+        elif isinstance(data, dict):
+            result = {}
+            for key, value in data.items():
+                result[key] = self.__prepare_data_for_printing(value)
+            return result
+
+        return data
+
+
+@dataclass
+class WebSocketRunnerStrings:
+    connecting = click.style('\t\tConnecting: ' + click.style('{url}', bold=True), fg='white')
+    abort = 'Connecting to {url} failed.'
+    success = click.style(f'\r{_SUCCESS} {{duration}}ms', fg='white')
+    failure = click.style(f'\r{_WARNING} {{duration}}ms', fg='white')
+    retry = click.style('\t\t  Retrying in {interval} seconds.', fg='white')
+
+
+class WebSocketRunnerLogger(WebSocketRunnerHooks):
+    def __init__(self):
+        self.__strings = WebSocketRunnerStrings()
+
+    def connecting(self, url: str):
+        print(self.__strings.connecting.format(url=url), end='')
+        sys.stdout.flush()
+
+    def abort(self, url: str):
+        print(self.__strings.abort.format(url=url))
+
+    def success(self, duration: int):
+        print(self.__strings.success.format(duration=duration))
+
+    def failure(self, duration: int):
+        print(self.__strings.failure.format(duration=duration))
+
+    def retry(self, interval_between_retries_in_seconds: int):
+        print(self.__strings.retry.format(interval=interval_between_retries_in_seconds))
+
+
+@click.group()
```

Comment:
I don't have a good solution, but to me it seems like everything below `def simulate()` should be in its own file as it really is just helping you test out everything else above

Reply:
I do not have a good idea to not add additional complexity. I can probably make that its own module or something like that. Mind if I do that in a followup ?

Reply:
Maybe add a TODO with a comment saying something like "Everything below this comment with a `click` decorator is for testing this file. It is here for reason xyz"